activation/activation.go

```diff
@@ -758,9 +762,12 @@ func (b *Builder) Regossip(ctx context.Context, nodeID types.NodeID) error {
 
 // SignAndFinalizeAtx signs the atx with specified signer and calculates the ID of the ATX.
 func SignAndFinalizeAtx(signer *signing.EdSigner, atx *types.ActivationTx) error {
-	atx.Signature = signer.Sign(signing.ATX, atx.SignedBytes())
+	// FIXME - there is no need to sign types.ActivationTX (only ActivationTxVx)
+	wireAtx := wire.ActivationTxToWireV1(atx)
+	atx.Signature = signer.Sign(signing.ATX, wireAtx.SignedBytes())
 	atx.SmesherID = signer.NodeID()
-	return atx.Initialize()
+	atx.SetID(types.ATXID(wireAtx.HashInnerBytes()))
+	return nil
 }
```

Comment:
I would leave this function as it is (maybe mark it deprecated) and change the `Builder` to sign the wire type directly during building. We want to avoid the conversion ATX -> wire.ATX completely; the builder should only know the wire type - the handler also only knows the wire type, verifies it and as the last step after successful verification convert it to the domain type and store it in the DB.

Reply:
In this PR I wanted minimal changes in Handler and Builder to make it work. I plan to refactor Builder and Handler in the next step in the following PRs.

Reply:
I added information about dependencies on the wire types to the issue about ATX versioning here: https://github.com/spacemeshos/go-spacemesh/issues/5774